subsys/bluetooth/services/ots/ots_dir_list.c

```diff
@@ -220,26 +191,72 @@ void bt_ots_dir_list_init(struct bt_ots_dir_list **dir_list, void *obj_manager)
 
 	memset(&dir_list_obj->metadata, 0, sizeof(dir_list_obj->metadata));
 	dir_list_obj->metadata.name = dir_list_obj_name;
-	dir_list_obj->metadata.size.alloc = sizeof((*dir_list)->_content);
+	dir_list_obj->metadata.size.alloc = DIR_LIST_MAX_SIZE;
 	dir_list_obj->metadata.type.uuid.type = BT_UUID_TYPE_16;
 	dir_list_obj->metadata.type.uuid_16.val = BT_UUID_OTS_DIRECTORY_LISTING_VAL;
 	BT_OTS_OBJ_SET_PROP_READ(dir_list_obj->metadata.props);
 
 	(*dir_list)->dir_list_obj = dir_list_obj;
 
-	/* Set size in dir_list_encode */
-	dir_list_encode(*dir_list, obj_manager);
+	dir_list_update_size(*dir_list, obj_manager);
 }
 
-ssize_t bt_ots_dir_list_content_get(struct bt_ots_dir_list *dir_list, void **data,
-				size_t len, off_t offset)
+ssize_t bt_ots_dir_list_content_get(struct bt_ots_dir_list *dir_list, void *obj_manager,
+				    void **data, size_t len, off_t offset)
 {
-	if (offset >= dir_list->net_buf.len) {
-		*data = NULL;
-		return 0;
-	}
+	struct bt_gatt_ots_object *obj;
+	int err;
+	size_t buffer_offset = 0;
+	size_t rec_len = 0;
+	size_t room = 0;
+	bool found = false;
+
+	err = bt_gatt_ots_obj_manager_first_obj_get(obj_manager, &obj);
+
+	__ASSERT(err == 0 && obj == dir_list->dir_list_obj,
+		 "Expecting first object to be the Directory Listing Object");
+
+	do {
+		/* keep resetting the buffer until we find the first object record that
+		 * is of intereset.
+		 */
```

Comment:
When is an object considered "of interest"? 
```suggestion
		/* keep resetting the buffer until we find the first object record that
		 * is of interest.
		 */
```

Reply:
Reworded this comment.